### DISCUSSION TITLE
Build error 'undefined' props from getStaticProps

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#24480] [Help]
When trying to access array in props, receiving error undefined.

```js
export default function Product(props) {
  const { name, description, slider } = props; // Props structure {name: String, description: String, slider: [{url: String}, {url: String},... }]}
  return (
    <div>
      <p>{name}</p>
      <p>{description}</p>
      {slider.map((image) => (
        <p>{image.url}</p>
      ))}
    </div>
  );
}

export async function getStaticProps({ params }) {
  const data = await getProductBySlug(params.slug); // Data from CMS
  const imageUrls = data?.[0].slider.map((image) => {
    console.log(image.url); // Successfully logs all urls to build console in Vercel
    return image.url;
  });
  return {
    props: {
      ...data?.[0],
    },
  };
}
```
The string type properties of 'name' and 'description' throw no error and are rendered.
The build throws no error if I use a ternary to check the presence of the prop first

```js
{slider ? slider.map((image) => (
        <p>{image.url}</p>
      )) : null}
```
and the elements are rendered.

Only one page uses this component, and the cms api function works fine.

**Comment by bobarros** [comment]
Are you sure that your images array is like these? It's not possible to repeat keys like that.

``` jsx
[
{url: String, url: String,... }
]
```

Should be something like:

``` jsx
 images: [
        { url: "String" },
        { url: "String" }
      ],
```

I tested here and worked fine:

``` jsx

<div>
          <p>{name}</p>
          <p>{description}</p>
          {images.map((image, index) => (
            <p key={image.url + index}>{image.url}</p>
          ))}
        </div>

export async function getStaticProps() {
  return {
    props: {
      name: "name sadjaskdl",
      description: "description askdasljj",
      images: [
        { url: "String" },
        { url: "String" }
      ],
    }
  }
}
```

**Reply by ghost** [reply]
Sorry, typo, structured like you pointed out. Edited.

**Reply by ghost** [reply]
logging props thus:

```js
export async function getStaticProps({ params }) {
  const data = await getProductBySlug(params.slug);

  const productData = data?.[0];
  console.log(JSON.stringify(productData));

  return {
    props: {
      ...data?.[0],
    },
  };
}
```

gives:

<img width="1222" alt="Screenshot 2021-04-26 at 23 54 45" src="https://user-images.githubusercontent.com/62337907/116155919-154faa80-a6eb-11eb-8ae3-f1058b9a6800.png">

Prop name here is different.

**Reply by bobarros** [reply]
I don't understand. This is your json:

```json
{
  "name": "name",
  "description": "description",
  "cover": {
    "url": "url 1"
  },
  "slider": [
    {
      "url": "url 2"
    },
    {
      "url": "url 3"
    }
  ]
}
```

There are no images. You should map the slider. No?

**Reply by ghost** [reply]
Yeah sorry, I renamed the prop in the original post, I mapped the slider array. Now updated.
Regardless, it's not working.

<img width="1072" alt="Screenshot 2021-04-27 at 00 59 40" src="https://user-images.githubusercontent.com/62337907/116161365-ee49a680-a6f3-11eb-8270-46645fa6c395.png">

Is this a Vercel issue?

**Reply by bobarros** [reply]
No, it's not. 
There's something that you're not seen it.

**Reply by ghost** [reply]
You're right, changing fallback to false in getStaticPaths has fixed the problem.
